compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/diagnostics/DiagnosticCodeTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerinalang.compiler.parser.test.diagnostics;
+
+import io.ballerina.compiler.internal.diagnostics.DiagnosticErrorCode;
+import io.ballerina.compiler.internal.diagnostics.DiagnosticWarningCode;
+import io.ballerina.tools.diagnostics.DiagnosticCode;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Test diagnostic codes.
+ *
+ * @since Swan Lake
+ */
+public class DiagnosticCodeTest {
+
+    private static final String DUPLICATE_DIAGNOSTIC_IDS_FOUND = "Duplicate diagnostic IDs found";
+    private static final String DUPLICATE_DIAGNOSTIC_MSGKEYS_FOUND = "Duplicate diagnostic messageKeys found";
+
+    @Test
+    public void testDiagnosticErrorCodesUniqueness() {
+        validateDiagnosticUniqueness(DiagnosticErrorCode.class);
+    }
+
+    @Test
+    public void testDiagnosticWarningCodesUniqueness() {
+        validateDiagnosticUniqueness(DiagnosticWarningCode.class);
+    }
+
+    // helpers
+    private <E extends DiagnosticCode> void validateDiagnosticUniqueness(Class<E> diagnosticCode) {
+        E[] diagCodes = diagnosticCode.getEnumConstants();
```

Comment:
I think getting information using `Class`, uses reflection.

Instead we can use something like: change the parameter to `DiagnosticCode[] codes`, and when parsing the codes, do: `DiagnosticCode[] codes = DiagnosticErrorCode.values();`

Reply:
I thought, parsing class would be better practice and need less codes. I'll change it